### ISSUE TITLE
Text is obscured at page breaks when printed 

### ISSUE THREAD

**Issue by dmgirard** [ZachSaucier/Just-Read#352]
**Describe the bug**
Text is obscured at page breaks when the adjusted web page is printed in Google Chrome. 

**To Reproduce**
1. Go to: https://www.psychologytoday.com/us/blog/the-right-mindset/202011/new-research-finds-pessimism-signals-bipolar-relapse?amp
2. Click on 'Just Read'
3. Click on 'Print article' button 
4. Scroll to last page of article on Print Preview.
5. Notice text obscured by page break.

**Expected behavior**
I expect that all text in the article will be clearly legible when I go to print the article. 

**Screenshots**
![Screenshot 2021-01-29 21_01_31](https://user-images.githubusercontent.com/78242890/106343859-8232e000-6275-11eb-83fa-91f80de5e826.png)

**Please complete the following information:**
 - OS: Microsoft Windows 10 
 - Browser: Google Chrome 
 - Version: Version 88.0.4324.104 (Official Build) (64-bit)

**Additional context**
None

**Comment by ZachSaucier** [comment]
Hmm. I'm not able to reproduce this issue. Using Chrome on Windows, same build.

Are you using the default stylesheet?

**Comment by dmgirard** [comment]
Yes. Just to be sure, I uninstalled the extension, reinstalled it, restarted the browser, and tried again. Same issue. I'm getting a ton of errors on this page when I click on 'Just Read.' See below. I don't know if that's relevant.

![Screenshot_2021-01-29 21_45_50](https://user-images.githubusercontent.com/78242890/106345080-8a8e1980-627b-11eb-9742-4a2c2b02f71a.png)

**Comment by ZachSaucier** [comment]
I'll try to test on another computer when I can but no promises on a fix. 

You can try adjusting the margins as a workaround. 

Those errors are not related to Just Read FYI.

**Comment by dmgirard** [comment]
Thanks. The problem occurs with the "Default" margin setting. When using custom margins (0.5" on top and bottom), the problem goes away. 

If I toggle the Headers/Footers setting, you can see what is happening more clearly. See the two images below. 

![2021-02-01 09_41_59-14 Proven Responses to the Most Frustrating Backtalk _ Empowering Parents](https://user-images.githubusercontent.com/78242890/106475417-e7ecbb00-6473-11eb-8d9f-3ffb6cfba9d2.png)

![2021-02-01 09_42_12-14 Proven Responses to the Most Frustrating Backtalk _ Empowering Parents](https://user-images.githubusercontent.com/78242890/106475476-f6d36d80-6473-11eb-915d-5d459ee0a10b.png)

**Comment by SaschaBur** [comment]
Same problem here on MAcOs Chrome. JustREad is GREAT, but this prevents me often from using it as i loose whole lines in the printout on PDF.

**Comment by ZachSaucier** [comment]
I still can't reproduce this issue. If you're able to help me reproduce it that'd be appreciated. Or if you edit the stylesheet to fix the issue, that'd be even better!

It's the same sort of issue as #51 which has been fixed (at least on my devices).

**Comment by Jillnvc** [comment]
I'm having this issue too and it's really frustrating that I didn't have it on the free version. I was better off not paying. I really liked Just Read and wanted to support it. 

I've tried every combination of margins and scale and it still cuts lines off at the top and bottom. (MacOS, Chrome). If I can fix it for the first page, it will cut off on later pages.

**Comment by ZachSaucier** [comment]
> I didn't have it on the free version

The free and paid versions are the exact same in this way. So it can't be an issue with you having Premium.

@Jillnvc Can you please share what URL you are trying to print?

**Comment by Jillnvc** [comment]
It happens for me with nearly every url. Here is the latest: https://austinchanning.substack.com/p/dear-nice-white-people

If you can't reproduce it with that, let me know and I'll post more.

**Comment by ZachSaucier** [comment]
I cannot reproduce the issue. But you do not need to share more pages. 

The issue must be due to some configuration or environmental setting. Given I can't reproduce the issue on any of the linked pages, it might be some time before someone posts enough info to where I can reproduce it in order to figure out a solution.

**Comment by Jillnvc** [comment]
Are you able to test it on a Mac with Chrome? If you can tell me what type of information I can offer, I will do it.

**Comment by ZachSaucier** [comment]
> Are you able to test it on a Mac with Chrome?

Yep. On Catalina. No issues.

> If you can tell me what type of information I can offer, I will do it.

Sorry, but I don't know what info might be relevant at this point.

**Comment by Jillnvc** [comment]
I'm still trying to figure this out. When I select the print command from within Just Read, it always resets to the default margins of 0. I can even print to PDF and then print to my printer and it will reset the default margins. And then, seemingly randomly, it will save my custom margins for the next thing, or maybe not. I need to change the default margins in the print settings. Is that part of Just Read or Chrome?

**Comment by ZachSaucier** [comment]
You can view the styles that Just Read adds for printing [here](https://github.com/ZachSaucier/Just-Read/blob/498d750d9a180a26a2888d7b6b0b908e44ffdcb9/required-styles.css#L222). You can override them in your personal theme stylesheets as needed.

**Comment by oep42** [comment]
Same problem here. The problem occurs with every Chromium browser I have tried.

Just Read 5.1.2
Windows 8.1
Browsers: 
--- Vivaldi 3.6.2165.40 (32-bit)
--- Slimjet 29.0.2.0 (32-bit)
--- Torch 69.2.0.1713 (32-bit)

The problem does not occur in the Firefox version of the extension.

**Comment by crogonint** [comment]
Hey Zach, ,this is James, I posted on the Chrome store about the issue..

I think what we are seeing is that Just Read is grabbing the page footer or lower margin from the original document and using it as a default for the printed output.. somehow I've noticed it doing it on web pages that use frames, and I THINK that that's what is going on.

Again, I totally blame Chrome, because Chrome's print driver OUGHT to know better than to hide text or content behind an empty margin! That said.. the footer/lower margin is useless anyway. I honestly don't know HOW many documents I saved before I noticed. I was so tickled with the "pretty print' style document in Just Read, that I never bothered checking that it was working in the Chrome print to PDF window.

**Comment by crogonint** [comment]
Well.. I've discovered that virtually no text from Github makes it in to Just Read. I've discovered that Reddit imports a pile of form columns in to Just Read and makes a bit of a mess of it. I've also discovered that the websites I seem to recall having issues with previously.. all appear to be working correctly right now. :/ Did you fix something??

I'll continue using Just Read for the time being, and report back when I find the websites that I was having problems with previously. :)

**Comment by ZachSaucier** [comment]
> I think what we are seeing is that Just Read is grabbing the page footer or lower margin from the original document and using it as a default for the printed output.. somehow

Hmm. I bring all the content into an iframe so I wouldn't think that would be an issue. But if you are able to test something and find this to be the case let me know!

> I've noticed it doing it on web pages that use frames, and I THINK that that's what is going on.

What do you mean by this?

> I've also discovered that the websites I seem to recall having issues with previously.. all appear to be working correctly right now. :/ Did you fix something??

If you mean from this morning to now, no 🙂

**Comment by crogonint** [comment]
For some reason, I never got a notice of your reply.. or I missed it.

If I recall, the problem appeared that Just Read was importing some sort of formatting from pages that use frames, and forwarding that to the Chrome Print window. I can't recall if it was margins or paragraph spacing or what, but I seem to recall that SOMETHING was making it past the Just Read window in to the Print window, and THAT was mucking up the margins and allowing text to get hidden behind the print margin.

Again again, I totally blame Chrome, because their print driver OUGHT to know better than to hide text!

I had been using Just Read and Print to PDF to save various wiki style articles for future reference. They pretty much ALL got hosed due to this issue, but when I went back to try them again, it was working properly.

The takeaway is that I noticed that Reddit and Github itself gave Just Read a nervous breakdown, as described above. I think if you concentrate on forcing those to pretty print correctly, you might come across the troublesome bit of code that is getting imported in to the CHrome Print window and causing all of the issues. :)

**Comment by ZachSaucier** [comment]
Closing this as I couldn't reproduce the issue. If someone runs into this issue let me know and I will try to reproduce it again
